_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,199 @@
+of a system test is to evaluate the end-to-end system specifications. Usually, the software is only one element of a 
+larger computer-based system. Ultimately, the software is interfaced with other software/hardware systems. System Testing 
+is actually a series of different tests whose sole purpose is to exercise the full computer-based system. (Guru)
+
+In our case the system tests validates the whole components/features which Strimzi provides. Everything is tested in 
+Kubernetes environment to fully represent production or user environment. Again to clarify I have an example: 
+
+```
+@Test
+void testReceiveSimpleMessage() {
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // Send messages to Kafka
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    // Deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // Deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
+}
+```  
+
+This test case as you can ensuring that we are able to receive messages from `KafkaBridge` component. In the setup phase 
+we deploy `cluster operator`, one node `Kafka cluster` and also the `KafkaBridge` itself. Everything is inside the real
+Kubernetes cluster. All test cases with related test suites you can find in the `system module` to make it more sense. What 
+is also worth to mentioned that our system test currently takes around ~30 hours but we have 3 separate sub-sets, which 
+runs in parallel so it is "only" approximately 10h per each sub-set.
+
+## How they run with relation with Kubernetes
+
+Basically, what I mentioned in previous section system tests run against real Kubernetes cluster. You can run these 
+tests against `Minikube`, big `Kubernetes` or `Openshift` cluster. In the `test module` we have our representation
```

Comment:
Maybe mention `minishift`? :smile:

Reply:
Not sure about that ... Minishift is OpenShift 3.11 and it's not so used as before now with OpenShift 4.x

Reply:
Okay, never mind :) it was just suggestion. I guess this way you can mentioned kind etc. and it's not worth